src/main/java/kr/hhplus/be/server/concert/service/ConcertRankingService.java

```diff
@@ -0,0 +1,123 @@
+package kr.hhplus.be.server.concert.service;
+
+import kr.hhplus.be.server.common.RedisKeyUtils;
+import kr.hhplus.be.server.concert.domain.SoldoutRank;
+import kr.hhplus.be.server.concert.repository.SoldoutRankJpaRepository;
+import kr.hhplus.be.server.queue.service.QueueService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.data.redis.core.ZSetOperations;
+import org.springframework.stereotype.Service;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+@Service
+public class ConcertRankingService {
+    private static final Logger log = LoggerFactory.getLogger(QueueService.class);
+
+    private final RedisTemplate<String, Object> redisTemplate;
+    private final SoldoutRankJpaRepository soldoutRankJpaRepository;
+
+    public ConcertRankingService(RedisTemplate<String, Object> redisTemplate, SoldoutRankJpaRepository soldoutRankJpaRepository) {
+        this.redisTemplate = redisTemplate;
+        this.soldoutRankJpaRepository = soldoutRankJpaRepository;
+    }
+
+    /**
+     * 매진 랭킹 조회 (상위 N개)
+     * @param limit 상위 N개
+     * @return 랭킹 리스트
+     */
+    public List<SoldoutRank> getTopRankings(int limit) {
+        try {
+            // 1. Redis 에서 상위 랭킹 조회
+
+            Set<ZSetOperations.TypedTuple<Object>> rankings =
+                    redisTemplate.opsForZSet().rangeWithScores(RedisKeyUtils.weeklyRankingKey(), 0, limit - 1);
+
+            List<SoldoutRank> result = new ArrayList<>();
+
+            // 2. Redis에서 반환되는 값 변환
+            for (ZSetOperations.TypedTuple<Object> tuple : rankings) {
```

Comment:
```suggestion
List<SoldoutRank> result = rankings.stream()
                .filter(this::isValidRankingTuple)  // 유효하지 않은 데이터 필터링
                .map(this::convertToSoldoutRank)    // 변환 로직 분리
                .collect(Collectors.toList());

private boolean isValidRankingTuple(ZSetOperations.TypedTuple<Object> tuple) {
    return tuple != null 
        && tuple.getValue() != null 
        && tuple.getScore() != null;
}
```

이런식으로 개선되면 코드가 더 깔끔해질것 같네요

`Object` 로 받는 부분도 개선되면 좋을것 같고요!